### DISCUSSION TITLE
Tenant ID - when importing

### DISCUSSION THREAD

**Discussion by MBstatsara** [Micke-K/IntuneManagement#83] [Q&A] [answered]
Hello

If we export some settings and reimport them to another tenant most are ok!
Occasionally we have some which are including tenant IDs in the export, and when imported again.. break!

Is there anyway i can edit these to not have a Tenant ID but change it to a variable, so that on import it uses that instead of the tenant ID of the last setup?

Thanks

**Comment by brucesa85** [comment]
> Hello
> 
> If we export some settings and reimport them to another tenant most are ok! Occasionally we have some which are including tenant IDs in the export, and when imported again.. break!
> 
> Is there anyway i can edit these to not have a Tenant ID but change it to a variable, so that on import it uses that instead of the tenant ID of the last setup?
> 
> Thanks

In this case, just open the exported file and change the tenant ID. :)

**Reply by MBstatsara** [reply]
Thanks - i know i can do it once, but as its importing is there not a variable that has the Tenant ID?
I don't want to keep having to do this every time, we are so near to automating a setup/check, id have to do each comparison as a different setting.
If there was only one it wouldn't be too much of an issue, but there is a few!

**Reply by Micke-K** [reply]
I can have a look at it. 
Do you gave an example of a profile with Tenant ID in the export?

Cheers!

**Reply by MBstatsara** [reply]
Sure!

Here it is - it sets onedrive to sign in as the user who is logged into the PC.

When you export it saves the TENANTIDHERE instead of that value.
Means that for some settings in intune, MS requires the Tenant ID - not sure why! but there are a few of these!

`{
  "Displayname": "Enable Onedrive Silent Logon and Known Folder Move",
  "Description": "This policy enables Onedrive Silent Logon and Known Folder move",
  "RAWJson": "{\"added\":[{\"enabled\":true,\"presentationValues\":[],\"definition@odata.bind\":\"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('9a4db949-29e4-4e31-a129-bf2b88d8fa1b')\"},{\"enabled\":true,\"presentationValues\":[{\"@odata.type\":\"#microsoft.graph.groupPolicyPresentationValueText\",\"value\":\"TENANTIDHERE\",\"presentation@odata.bind\":\"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('39147fa2-6c5e-437b-8264-19b50b891709')/presentations('fbefbbdf-5382-477c-8b6c-71f4a06e2805')\"},{\"@odata.type\":\"#microsoft.graph.groupPolicyPresentationValueText\",\"value\":\"0\",\"presentation@odata.bind\":\"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('39147fa2-6c5e-437b-8264-19b50b891709')/presentations('35c82072-a93b-4022-be14-8684c2f6fcc2')\"}],\"definition@odata.bind\":\"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('39147fa2-6c5e-437b-8264-19b50b891709')\"},{\"enabled\":true,\"presentationValues\":[],\"definition@odata.bind\":\"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('81c07ba0-7512-402d-b1f6-00856975cfab')\"},{\"enabled\":true,\"presentationValues\":[],\"definition@odata.bind\":\"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('61b07a01-7e60-4127-b086-f6b32458a5c5')\"},{\"enabled\":true,\"presentationValues\":[],\"definition@odata.bind\":\"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('8866d7bd-42fb-4695-b6f2-80e0a90b1ac3')\"},{\"enabled\":true,\"presentationValues\":[],\"definition@odata.bind\":\"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('f974758d-1fab-42fe-ad36-3a6cd25c49c1')\"}],\"updated\":[],\"deletedIds\":[]}\r\n",
  "Type": "Admin",
  "GUID": "7b41924e-3051-4a23-b0d0-8tfwvef3d16a"
}`

**Comment by Micke-K** [comment, marked as answer]
Hello!

Here is a new file for you to test.

Unzip and replace the MSGraph.psm1 file in the IntuneManagement\Extensions folder
Restart app
Try export and then import in another tenant

Let me know how it goes...

Cheers!

[MSGraph.zip](https://github.com/Micke-K/IntuneManagement/files/9431867/MSGraph.zip)

**Reply by MBstatsara** [reply]
Awesome that worked really well!
Will that happen for everywhere it shows "org ID" in the export?

Thanks so much

**Reply by Micke-K** [reply]
No problem!

The code will update any exported profile that has the actual tenant id configured in it.

Cheers!

**Comment by MBstatsara** [comment]
Will this be wrapped into the current next release?

**Reply by Micke-K** [reply]
Yes, I've just been super busy last month so I haven't had time to prepare the release. I always have a few checks I do before a release.

**Comment by MBstatsara** [comment]
This seems to be broken in the last two updates
It no longer changes the Tenant Id to the new Tenant ID - keeps the old exported Tenant ID present.
If i copy my msgraph file back from this discussion to the new one it still doesnt work, so something else in the import settings must have changed.
Im now back using release 3.7.1 with the graph ps1 from this discussion.

**Reply by MBstatsara** [reply]
Nope - seems like rolling back doesnt work either!
I tried exporting it using the old version and importing it again too and it doesnt change the tenant ID..

Shall i try on a PC thats not been affected by the installs before? I.e. using the PS1 from this script or a brand new VM with the latest version?
Its strange its affected the old version after i updated to the fixed version and the latest one too

**Comment by Micke-K** [comment]
Hello,

I'll have a look at this and see what it could be. 

Cheers!

**Reply by Micke-K** [reply]
I can't see how this could have worked at all :)

I just realised that your json example was for Administrative Templates and I didn't add the fix for them. It should work if you set it with a Setting Catalog. I need to build a generic update for this so it works with all the weird special solutions that is in. The code. 

I'll see when I can get time to fix this but I'll put it on the priority list...

Cheers!

**Comment by Micke-K** [comment]
Hello,

Can you try this? I've changed so all Save and Load will use a common function.

Cheers

[EndpointManager.zip](https://github.com/Micke-K/IntuneManagement/files/9877261/EndpointManager.zip)

**Reply by MBstatsara** [reply]
Awesome! that seems to be working well now!
I'll try a few more tenants just to make sure, but i think we have it sorted!
